config/test.exs

```diff
@@ -41,7 +41,7 @@ config :crawler, Crawler.Mailer, adapter: Swoosh.Adapters.Test
 # Print only warnings and errors during test
 config :logger, level: :warn
 
-config :crawler, Oban, crontab: false, queues: false, plugins: false
+config :crawler, Oban, testing: :inline
```

Comment:
Sorry, I quite can't get this `we just test it trigger the Oban job`. As I know so far, Oban is automatically triggered after we create job and insert to the DB.

Reply:
If we do not allow the Oban to start the job when we insert it into the DB, it will be hard to control later on, especially when our codebase is quite large; we would need a lot of `use_cassette` for example

```
it 'this test will trigger job a`
  resource_a = xx

  use_cassette a
end

it 'this test will trigger job b`
  resource_b = xx

  use_cassette b
end

it 'this test will trigger 2 different jobs`
  resource_a = xx
  resource_b = xxx

  use_cassette a_b (this is a separate use_cassette file contains both a and b)
end
```

So for that, in the actual project, we do not use the `testing inline` like that, instead we do not let the Oban start the job, we just use the [assert_enqueued](https://hexdocs.pm/oban/Oban.Testing.html)

```
it 'this test will enqueued job a`
  resource_a = xx

  assert_enqueued a
end

it 'this test will enqueued job b`
  resource_b = xx

  assert_enqueued b
end

it 'this test will enqueued 2 different jobs`
  resource_a = xx
  resource_b = xxx

    assert_enqueued a
    assert_enqueued b
end
```

and in the `job a, job b` we test it isolates, like what you did for the `test/crawler_worker/keyword/crawler_worker_test.exs`

Reply:
@andyduong1920 Thanks for the detail. I fixed in [816d498](https://github.com/vnntsu/elixir-ic/pull/39/commits/816d49800d958810db07c5f0df155ef48e02fc55), it consists of Oban config to `manual` instead of `inline` and remove `use_cassette` from normal test cases.